test/model/database_test.dart

```diff
@@ -330,6 +402,56 @@ void main() {
     });
 
     // TODO(#1593) test upgrade to v9: legacyUpgradeState set to noLegacy
+
+    test('upgrade to v10: with accounts available, '
+        'insert first account id as the last-visited account ID', () async {
+      final schema = await verifier.schemaAt(9);
+      final before = v9.DatabaseAtV9(schema.newConnection());
+      await before.into(before.accounts).insert(v9.AccountsCompanion.insert(
+        realmUrl: 'https://chat.example/',
+        userId: 1,
+        email: 'asdf@example.org',
+        apiKey: '1234',
+        zulipVersion: '10.0',
+        zulipMergeBase: const Value('10.0'),
+        zulipFeatureLevel: 370,
+      ));
+      await before.into(before.accounts).insert(v9.AccountsCompanion.insert(
+        realmUrl: 'https://example.com/',
+        userId: 2,
+        email: 'jkl@example.com',
+        apiKey: '4321',
+        zulipVersion: '11.0',
+        zulipMergeBase: const Value('11.0'),
+        zulipFeatureLevel: 420,
+      ));
+      await before.close();
+
+      final db = AppDatabase(schema.newConnection());
+      await verifier.migrateAndValidate(db, 10);
+      await db.close();
+
+      final after = v10.DatabaseAtV10(schema.newConnection());
+      final intGlobalSettings = await after.select(after.intGlobalSettings).getSingle();
+      check(intGlobalSettings.name).equals(IntGlobalSetting.lastVisitedAccountId.name);
+      check(intGlobalSettings.value).equals(1);
```

Comment:
Instead of a hard-coded 1, it's better if this refers to the account ID of the intended one out of the two accounts inserted above.

Could do that either by selecting the account back out of the table, or using the return value of `insert`.